zebra-state/src/service/arbitrary.rs

```diff
@@ -151,6 +148,7 @@ pub(crate) fn partial_nu5_chain_strategy(
 
 /// Return a new `StateService` containing the mainnet genesis block.
 /// Also returns the finalized genesis block itself.
+#[cfg(test)]
```

Comment:
It's a bit confusing mixing `#[cfg(test)]` and `#[cfg(any(test, feature = "proptest-impl"))]` code in the same module.

Can you move these functions and their imports to a new `tests::setup` module instead?